### DISCUSSION TITLE
Extracting and using intrinsic growth rate (r) from FishLife

### DISCUSSION THREAD

**Discussion by mcleamj** [James-Thorson-NOAA/FishLife#26] [Q&A]
Hi James et al.

I would like to extract intrinsic growth rate (r) values from FishLife for several species to be used in some basic Schaefer surplus production models, for which I have data on all other parameters. However, in reading through a FishLife vignette I came across the following text:

_Finally generation time `G` and intrinsic growth rate `r` are calculated based on a nonlinear transformation of other traits using the Euler-Lotka formula, and we therefore report the mean in either log-space or natural space, obtained by sampling from the constituent traits, calculating the values for each sample, and then computing the mean of those calculations.  Given this procedure, exponentiating `ln_G` will not typically equal `G` and the same holds for `ln_r` and `r`._  

Thus, I am wondering if there is indeed a proper way to extract "true" r values and if this is species dependent. 

Thanks very much!

**Comment by James-Thorson-NOAA** [comment]
Thanks for posting on the Discussion board!  

A few context questions:
- Does your application use a point-estimate or permit using the full distribution (e.g., via an [ensemble](https://academic.oup.com/icesjms/article-abstract/76/4/870/5315645))?
- What is your use-case generally speaking?

If using a point-estimate, I generally recommend the mean of r (not log_r).  If using a distribution, I think it makes sense to explore the distribution resulting from the log_r mean and SD, but also with some caution about the tails (which might be poorly approximated and quite low/high)

**Reply by mcleamj** [reply]
Thank you - my application will use a point estimate. The idea is to use a simple surplus production model to evaluate biomass changes under different levels of catch for several species. The aim is to use the r value from FishLife as a parameter in this model. Thanks very much!

**Comment by James-Thorson-NOAA** [comment]
Makes sense! Yeah mean of r (not ln_r) makes sense absent other
considerations.

On Wednesday, February 28, 2024, Matthew McLean ***@***.***>
wrote:

> Thank you - my application will use a point estimate. The idea is to use a
> simple surplus production model to evaluate biomass changes under different
> levels of catch for several species. The aim is to use the r value from
> FishLife as a parameter in this model. Thanks very much!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/James-Thorson-NOAA/FishLife/discussions/26#discussioncomment-8625046>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AL62VMVX53FAZKCVQOFXR4DYV7PGBAVCNFSM6AAAAABD4O7XK2VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DMMRVGA2DM>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***
> com>
>


-- 
James Thorson (he/him)

Alaska Fisheries Science Center, NMFS
Affiliate Faculty, University of Washington

**Reply by mcleamj** [reply]
Thanks very much - so the mean point estimates directly from the plot_taxa function will give the correct value?

**Comment by James-Thorson-NOAA** [comment]
Correct.

On Friday, March 1, 2024, Matthew McLean ***@***.***> wrote:

> Thanks very much - so the mean point estimates directly from the plot_taxa
> function will give the correct value?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/James-Thorson-NOAA/FishLife/discussions/26#discussioncomment-8645497>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AL62VMT5D6FP6KGUXIOZSMLYWCWHRAVCNFSM6AAAAABD4O7XK2VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DMNBVGQ4TO>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***
> com>
>


-- 
James Thorson (he/him)

Alaska Fisheries Science Center, NMFS
Affiliate Faculty, University of Washington

**Reply by mcleamj** [reply]
Thanks a lot!!
